app/controllers/stafftools/users_controller.rb

```diff
@@ -0,0 +1,29 @@
+module Stafftools
+  class UsersController < StafftoolsController
+    before_action :set_user, only: [:impersonate]
+    skip_before_action :authorize_access, only: [:stop_impersonating]
+
+    def impersonate
+      if @user
+        session[:impersonated_user_id] = @user.id
+        redirect_to root_path
+      else
+        flash[:error] = 'This user does not exist on Classroom for GitHub'
+        redirect_to :back
+      end
+    end
+
+    def stop_impersonating
+      not_found unless true_user.try(:staff?)
```

Comment:
I know returning `404` helps with avoiding disclosure, but since this app is open source is there any reason not to just return unauthorized here?